src/code/z_actor.c

```diff
@@ -1570,7 +1570,14 @@ u32 Actor_HasParent(Actor* actor, PlayState* play) {
     }
 }
 
-s32 func_8002F434(Actor* actor, PlayState* play, s32 getItemId, f32 xzRange, f32 yRange) {
+/**
+ * Allows to pick up an item (GetItem or GI), lift an actor or catch various actors in bottles
```

Comment:
```suggestion
 * Allows to pick up an item (GetItem), lift an actor or catch various actors in bottles
```
GI and GetItem are the same right? Or is this just here to help with the abbreviation

Reply:
I think we need to workshop the wording of this comment in general more. I tried writing a new one, but the exact words arent coming to me at the moment. The current wording is still as if its called "Actor_PickUp", so we need to get into a bit more of the intricacies of an "offer" and what that means for the process of getting an item

Reply:
my attempt:
(current comment)
```c
 * Allows to pick up an item (GetItem or GI), lift an actor or catch various actors in bottles
 * within the specified range.
 *
 * GI_NONE is usually used as a special case to lift an actor
 * GI_MAX is usually used to catch an actor in a bottle
```
->
```c
 * This function covers various interactions with the player actor, using Get Item IDs (see `GetItemID` enum).
 * It is typically used to give items to the player, but also has other purposes.
 *
 * This function does not perform the requested action itself, it only carries the request onto the player actor if context allows it (e.g. the player is in range and not busy with certain things).
 * Hence its name using "Offer", as it is up to the player actor to honor the request.
 *
 * This function handles most requests the same, besides a few checks.
 * The following description of what the `getItemId` values can do is provided here for completeness, but these behaviors are entirely out of the scope of this function. All behavior is defined by the player actor.
 *
 * Positive values (`GI_NONE < getItemId < GI_MAX`): give an item to the player. The player may not get it immediately (for example if underwater), but is expected to in a near future.
 * Negative values (`-GI_MAX < getItemId < GI_NONE`): used by treasure chests to indicate the chest can be opened (by pressing A)
 * `GI_NONE`: allows the player to pick up the actor (by pressing A), to carry it around.
 * `GI_MAX`: allows the player to catch the actor in a bottle.
```

Reply:
yea this is a good start

- I find the `GI_NONE < getItemId < GI_MAX` notation kinda confusing and feel it can be omitted, but that may just be me
- in a near future -> in the near future
- for chests, we can try to fit in that its the negative version of an otherwise positive get item id. idk how to word it well
- "Allows the player to catch *specific actors* in a bottle", just want to stress that not any actor can be caught in a bottle, player has to have implemented it for specific actors on his end
- `@return true if Player has passed all checks and is currently able to accept an offer, false if not` or something like that

Reply:
Not sure about `GI_NONE < getItemId < GI_MAX`, I feel like it's needed to exclude these specific `GI_` values, since they're otherwise part of Get Item IDs
Also shows `-GI_MAX` does nothing (wasted opportunity for Morita there to map a fishing thing to it)

2nd draft taking into account other remarks
```
 * This function covers various interactions with the player actor, using Get Item IDs (see `GetItemID` enum).
 * It is typically used to give items to the player, but also has other purposes.
 *
 * This function does not perform the requested action itself, it only carries the request onto the player actor if
 * context allows it (e.g. the player is in range and not busy with certain things).
 * Hence its name using "Offer", as it is up to the player actor to honor the request.
 *
 * This function handles most requests the same, besides a few checks.
 * The following description of what the `getItemId` values can do is provided here for completeness, but these
 * behaviors are entirely out of the scope of this function.
 * All behavior is defined by the player actor.
 *
 * Positive values (`GI_NONE < getItemId < GI_MAX`):
 *   Give an item to the player. The player may not get it immediately (for example if underwater), but is expected to
 *   in the near future.
 * Negative values (`-GI_MAX < getItemId < GI_NONE`):
 *   Used by treasure chests to indicate the chest can be opened (by pressing A).
 *   The item gotten corresponds to the positive Get Item ID `abs(getItemId)`.
 * `GI_NONE`:
 *   Allows the player to pick up the actor (by pressing A), to carry it around.
 * `GI_MAX`:
 *   Allows the player to catch specific actors in a bottle.
 *
 * @return true If the player actor is in range, seems in capacity of handling the offer, and the offer has been passed on to it. 
```

Reply:
It's still a bit more wordy than it needs to be, and the first line really needs to cut straight to the point quicker... right now the wording of the first line is too nebulous.

I'd say the core purpose is to send a message to the player that an "item" can be obtained, with the return value being true if the message was properly sent. We should lead with that concept in the first sentence, and bang out the details later. Similarly, the return description should be simplified to just `@return true if the message was passed to the player`, because the checks performed are complex and are better expressed by the code itself.

Paragraphs 2 and 3 are filled with needless fluff, especially if you restructure with the core purpose above in mind. I was able to cut it down to just these lines:

    This function passes the request onto the player actor only if context allows it (e.g. the player is in range and not busy).
    The following description of what the `getItemId` values can do is provided here for completeness

Reply:
I do agree it can be trimmed down a bit. But some of the details youre cutting out are still important. it doenst need to be that succinct. This is a complicated interaction, so we can spare a few words

when I can, I will make my own edits to our current version of the comment